2017-2018/GameEngine/vision/__init__.py

```diff
@@ -0,0 +1,10 @@
+
+try:
+    # If the prerequisites for the cv module are missing it will use the
+    # keyboard shim
+    from .cv import MovementProcessor
+except ImportError:
+    from .shim import MovementProcessor
```

Comment:
Instead of this, could we just have the 2 movementProcessors coexist? Even after we've built the CV module, we might want to use the keyboard simulator for testing

Reply:
Definitely can, just have to define a method for manually selecting which one. My preferred method is environment variables, then you just run it as `KEYBOARD=1 ./cameraReader.py` instead of `./cameraReader.py` for example

Reply:
I was thinking about just having a separate script for the keyboard processor. There's no need to couple it with cameraReader

Reply:
Okay, as long as one never tries to run multiple input scripts at the same time. The way the server is currently implemented, it does not distinguish between different input clients, so worst case the bytes of messages from the two clients could be intertwined and break it. It could be rewritten to support multiple clients, but I don't think that's necessary.